### PR TITLE
RD2: Handle Cancel and Save buttons on save/error

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
- Display an error when the user is missing field permissions and disable the [Save] button. 
- Disable [Cancel] and [Save] buttons when the record is being saved.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
Tracking [W-7865067](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008TJt8IAG/view)

**Review by force2b — APPROVED** [review]

File: src/lwc/rd2EntryForm/rd2EntryForm.js [hunk 1]
```diff
@@ -480,36 +482,31 @@ export default class rd2EntryForm extends LightningElement {
...
+            : constructErrorMessage(error);
 
-        this.handlePageOnError(disableSaveButton);
+        this.setSaveButtonDisabled(true);
     }
 
     /***
     * @description Handle component display when an error occurs
-    * @param disableSaveButton: Indicates if the Save button should be disabled
+    * @param isDisabled: Indicates if the Save button should be disabled
```
Comment (npsp-reedestockton) [inline src/lwc/rd2EntryForm/rd2EntryForm.js:502]: This is a real nitpick, but isDisabled sounds like the button is already disabled, however that is the result of this action.  I'm not sure that isDisabled is an improvement on disableSaveButton.  

That is a very minor quibble and I have no problem with leaving it as is, so I don't plan to ask for changes.

**Review by npsp-reedestockton — APPROVED** [review]

File: src/lwc/rd2EntryForm/rd2EntryForm.js (hunk 1 above)
Comment (RadSFDeveloper) [reply src/lwc/rd2EntryForm/rd2EntryForm.js:502]: The input is a Boolean so it should start with `is`, `has`, `should`. Also, the previous code version statement `const disableBtn = !!(disableSaveButton && disableSaveButton === true);`  is confusing. I did leave the `disableBtn` var name, although it is more of a method name not a var name.

**Comment by RadSFDeveloper** [comment]
@rjanjanam , the PR has merge conflicts. I am going to resolve it now, but once it is resolved in a new commit, the code needs to be checked out and the dev_org flow re-run on the same NPSP org.
CC @rponti-sforg , @ravensisco

**Comment by RadSFDeveloper** [comment]
@rjanjanam , the merge conflict has been resolved. Can you please re-deploy code onto the existing NPSP scratch org? Thanks.

**Review by andrewyu-salesforce — APPROVED** [review]
The disable button looks good. Good to know that conditional button rendering does not work in FireFox.

**Pull request merged**